src/ducks/files/RenameInput.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import FilenameInput from '../../components/FilenameInput'
+import { getUpdatedName, rename, updateFileName, abortRenaming } from './rename'
+
+const RenameInput = ({ name, updateRenaming, rename, abortRenaming }) => (
+  <FilenameInput name={name} onChange={updateRenaming} onSubmit={rename} onAbort={abortRenaming} />
+)
+
+const mapStateToProps = (state, ownProps) => ({
+  name: getUpdatedName(state)
+})
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  updateRenaming: name => dispatch(updateFileName(name)),
+  rename: () => dispatch(rename()),
+  abortRenaming: name => dispatch(abortRenaming())
```

Comment:
Not sure the `renaming` workflow should be hosted in a stateful component, not in the application's global state. Cf. [“You Might Not Need Redux”](https://medium.com/@dan_abramov/you-might-not-need-redux-be46360cf367)